client/my-sites/themes/theme-options.js

```diff
@@ -113,7 +120,10 @@ const tryandcustomize = {
 			// is less readily available since it needs to be fetched using the `QuerySitePlans` component.
 			( isJetpackSite( state, siteId ) && ! isPremiumThemeAvailable( state, themeId, siteId ) )
 		) ||
-		( isJetpackSite( state, siteId ) && ! isThemeAvailableOnJetpackSite( state, themeId, siteId ) )
+		( isJetpackSite( state, siteId ) && ! isThemeAvailableOnJetpackSite( state, themeId, siteId ) ) ||
+		( config.isEnabled( 'jetpack/pijp' ) &&
+			isJetpackSite( state, siteId ) &&
+			! hasFeature( state, siteId, FEATURE_UNLIMITED_PREMIUM_THEMES ) )
```

Comment:
This won't work, because isPremiumThemeAvailable checks isThemePurchased and that checks for purchases, specifically, the product `premium_theme` and we don't have that one in Jetpack sites because themes are acquired by plan.